### PR TITLE
fix(firewall rule): avoid errors and set null values to rule attributes

### PR DESCRIPTION
**Pull request by Jisay**

### CODE REVIEW

File: client/app/ip/ip/firewall/rule/add/ip-ip-firewall-rule-add.controller.js [hunk 1]
```diff
@@ -77,6 +77,20 @@ angular.module("Module.ip.controllers").controller("IpFirewallAddRuleCtrl", ($sc
 
     $scope.addRule = function () {
         $scope.loading = true;
+
+        // set empty string to null values to avoid API error
+        if ($scope.rule.source.trim() === "") {
+            $scope.rule.source = null;
```
Comment (FredericEspiau) [inline client/app/ip/ip/firewall/rule/add/ip-ip-firewall-rule-add.controller.js]: `$scope.rule.source = _.isEmpty($scope.rule.source) ?  null : $scope.rule.source`

Comment (Jisay) [reply client/app/ip/ip/firewall/rule/add/ip-ip-firewall-rule-add.controller.js]: :-) Good too. But `_.isEmpty(" ")` return `false` ;-)

File: client/app/ip/ip/firewall/rule/add/ip-ip-firewall-rule-add.controller.js [hunk 2]
```diff
@@ -12,7 +12,7 @@ angular.module("Module.ip.controllers").controller("IpFirewallAddRuleCtrl", ($sc
     };
 
     $scope.resetOptionField = function () {
-        if ($scope.rule.protocol !== "TCP") {
+        if ($scope.rule.protocol !== "tcp") {
```
Comment (FredericEspiau) [inline client/app/ip/ip/firewall/rule/add/ip-ip-firewall-rule-add.controller.js]: What about a service which would check if the value is empty and doesn't care if it's uppercase or lowercase ?

Comment (jleveugle) [reply client/app/ip/ip/firewall/rule/add/ip-ip-firewall-rule-add.controller.js]: it's going to be refacto @FredericEspiau, Fabien found a new bug ;)

**Pull request merged**